### DISCUSSION TITLE
Release 1.0

### DISCUSSION THREAD

**Discussion by Noordfrees** [widelands/widelands#4585] [General]
We have so many new features that we don't need many more new ones until release 1.0. In my opinion, we should gradually start entering bug-hunting mode.

I have a few more features in review that I want to see in the release: Configurable hotkeys; choosable UI themes; fri03; targeted scouting. 
And I'm intending to (complete and) polish those of my features that are not 100% complete yet: Add-on system, main menu redesign, amazon graphics. The training wheels are another feature in this category.
Larger new features (multithreading, scenario editor, naval warfare, animal herds simulation) are candidates for postponing to release 1.1. I'm going to go through the wishlist soon and reschedule any issues that are not important for the release.

Should we already set a date for feature freeze? Perhaps the first of March, so we still have enough time to implement some new stuff, and with the understanding that existing features can still be polished and improved during the freeze. Opinions?

**Comment by Noordfrees** [comment]
I have gone through the open PRs and issues and either rescheduled them or marked them as urgent: https://github.com/widelands/widelands/milestones/v1.0

**Comment by klaus-halfmann** [comment]
I vote +1 for getting the multithreading in. It _will_ cause more bugs but open the door for better perfomance on
current CPUs.

**Reply by Noordfrees** [reply]
Okay
I have merged master there so the PR is ready again. If it is merged this month, let's have it in 1.0 (any later and there would be too little time for my liking to discover nasty, release-critical bugs though)

**Reply by frankystone** [reply]
There are no performance problems, at least in release builds.

I vote for having multithreading in v1.1. There are a lot of new features anyway.

**Reply by Noordfrees** [reply]
On second thoughts, I now fix my mind on postponing multithreading to 1.1. Adding a big feature that can potentially introduce several new bugs is awkward shortly before a release, but doing it before the "first-ever stable release"… let's get it in shortly *after* the release instead to be safe.

**Reply by klaus-halfmann** [reply]
Yes get it in early in 1.1. and then wait for all the bugs,

**Comment by Noordfrees** [comment]
@widelands/developers @widelands/elders @widelands/chieftains 

## Announcement

The First Snow Feature Freeze for the release of Widelands 1.0 will take place on Monday, the first of March 2021.

First-Snow feature freeze means:
-    no new features
-    no style fixes
-    no changes to campaigns
-    no new tagged strings
-   only minimal invasive bug fixes. That is if a bug can be fixed in a dirty-but-minimal-invasive change and in a correct way, the one that doesn't touch on any other code should be preferred and the correct way should go into a feature branch.

Bugs may still be fixed and existing features can still be polished and improved during the freeze.
As soon as all targeted bugs are fixed, the release schedule will be announced.

**Comment by hessenfarmer** [comment]
If we really go into feature freeze on 1st march we should remove the training wheels code imho, as they are implemented only as proof of concept and not anyway near a mature level neither are they far from being complete and so would just lead to confusion.

**Reply by Noordfrees** [reply]
Perhaps we'll get more wheels until the release in which case they can stay in. But I guess this is not likely so +1 for commenting out the relevant interface parts. Will do so next month

**Comment by Noordfrees** [comment]
#4458 is a big PR which has lots of merge conflicts currently and quite a number of review comments that need addressing. I would be in favour of postponing it to 1.1, except that it fixes a b21 savegame loading bug that must be taken care of for 1.0. Would someone like to try if #4273 could be fixed in a less involved way, so we could have the minimalistic fix now and postpone the big refactoring?

**Reply by matthiakl** [reply]
I started looking into it a while ago and can investigate now, if I can bring it to a valid fix.

**Comment by Noordfrees** [comment]
Would anyone like to take care of updating the changelog? 
(The last update, 987bc2153cfc94d01f653da569e8c3a0236ff981, was up to commit 4585505, 8 months ago.)

**Reply by Noordfrees** [reply]
Since nobody else seems to want this most enjoyable and popular task I'll do it

edit: branch at https://github.com/Noordfrees/widelands/tree/changelog

**Reply by frankystone** [reply]
For several weeks i have started to work on this also. But for some reason my changes got lost... and i didn't wanted to start from scratch :-(

Thanks for taking this task!

**Comment by Noordfrees** [comment]
### Winter Time Feature Freeze Coming Soon

We're now down to 1 bug and 1 graphics-only enhancement waiting on the [milestone](https://github.com/widelands/widelands/milestones/v1.0). As soon as these few things (and any other important issues that might pop up in the next days) have been taken care of, we'll call the [Winter Time Feature Freeze](https://www.widelands.org/wiki/ReleasingWidelands/#step-2-winter-time-feature-freeze-the-real-feature-freeze).

This feature freeze extends to everything: from translatable strings to media, code, style. So there is a time of three weeks where everything should already be perfectly working before the release candidate, to make sure that no last-minute things come up. Translations will still be pulled though; the validator will frequently be run and double-checked.
Obviously, new surfacing critical bugs are to be vanquished. Non-critical bugs are to be deferred to the next version.

This freeze will last for approximately **3 weeks**; then the release candidate will be released. One week after the RC, the actual release will follow.

**Reply by Noordfrees** [reply]
And now a new desync was found: https://github.com/widelands/widelands/issues/4816
The winter time freeze may still have to wait a bit longer.

**Reply by Noordfrees** [reply]
Only two more codechanges for v1.0 left to review currently – come on everyone :)
#4826 (a one-liner fix for a small but irritating glitch)
#4827 (a collection of fixes for several evil crashes related to add-ons)

**Reply by Noordfrees** [reply]
Now that all targeted bugs and features except for one graphics-only change have been merged, it's time to schedule the **winter-time freeze** for

## ❄️❄️❄️ Saturday, the 15th of May 2021 ❄️❄️❄️

From then on, only fixes for newly found critical bugs (and the targeted #4785 & #4820) may still be merged.

The winter-time freeze will hold for several weeks to allow extensive testing. The date for the **release candidate** will be determined in accordance with the [Tournament](https://www.widelands.org/forum/topic/5225/)'s schedule for maximum testing coverage. The **actual release** is expected **1 week** after the RC.

**Comment by klaus-halfmann** [comment]
Thanks to @Noordfrees and all our developers for you constant efforts for our Project!

When do we intend to create the actual Release-Branch?
AND we should provide a more prominent way to download a "latest" build.
I was talking with some casual players, but is quite complex for them to download such a version.

**Reply by Noordfrees** [reply]
I'll create the release branch on the release candidate day (or at most 1 day before that; no earlier, as it would unnecessarily complicate bugfix backporting).

We provide development build download links on the [Downloads](https://www.widelands.org/wiki/Download/#unofficial-development-builds) page, which offers direct links to AppVeyor and MacOS binaries as well as commandline instructions for various distros. It's a wiki page – if you see room for improvements feel free to update it :)

**Comment by Noordfrees** [comment]
# Widelands Tournament 2021 

As we are reaching towards a new release, @hessenfarmer is hosting a tournament to test the current build for its suitability in tournament mode.

Subscriptions are now open. Please subscribe in the [tournament thread](https://www.widelands.org/forum/topic/5225) or by PM to [hessenfarmer](https://www.widelands.org/profile/hessenfarmer/). Planned start is mid-May 2021.

The Tournament 2021 will be played using add-ons with a very recent development build announced shortly before the beginning. The version of the game might be updated round by round if bugs get reported and fixed during the tournament.

It will be played on a version of the Colonies map with no coal and no direct contact possible. The win condition will be to create 10 fully trained soldiers and store them in the warehouses in the shortest time.

The Tournament will be played in 5 rounds 1v1. Each round uses a different tribe: Amazons, Atlanteans, Barbarians, Empire, and Frisians.

## [For more information, visit the forum thread here.](https://www.widelands.org/forum/topic/5225)

**Comment by Noordfrees** [comment]
# Widelands 1.0 Release Schedule

The first release candidate will be published on **Tuesday, June 8, 2021**.
The actual release will follow one week later on **Tuesday, June 15, 2021**.

**Comment by frankystone** [comment]
One PR left... this will be one of the most exiting releases! 

And it ensures the work will never end :) (that's the purpose of this incredible project)

**Reply by Noordfrees** [reply]
Six days to go :) 
Anyone to review #4902? Please?

**Comment by hessenfarmer** [comment]
I wrote a PM in the forum to @tibor95 asking for some AI Training, but got no feedback yet.  
Although this is not super important some rouinds would be very helpful I guess as we changed some parts of the code and have a new tribe

**Comment by Noordfrees** [comment]
Oh wonderful. The GitHub-provided .tar.gz is corrupt (the .zip works though). And the curl website has chosen NOW out of all inconvenient times to be temporarily (I hope) unavailable. At least the x64 windows build is completed, I hope the other one will be able to rerun successfully some time later…
edit: curl is back online 😌

**Reply by tothxa** [reply]
The tar.gz also worked for me, so I try to use it for the flatpak...

**Reply by Noordfrees** [reply]
I tried downloading it again several times but every time gzip claims it's an invalid file…

I'm wondering which year it will be by the time the windows binaries finish uploading, launchpad is sloooow today 
🐢 > 🐌 > 🦥 >> ![](https://launchpad.net/@@/launchpad-logo-and-name-hierarchy.png)

**Reply by Noordfrees** [reply]
AAAaaargh!! My win32 upload which has been running for the past TEN HOURS and was just nearing completion was cancelled by launchpad due to some supposed site update of theirs!
I'll restart it, but I have to correct my above statement – I no longer wonder which year the uploads are done but which century.

**Reply by TinoM** [reply]
Maybe release on Github? Or also upload the binaries here:
https://github.com/widelands/widelands/releases/tag/v1.0-rc1

=> Edit release

**Reply by Noordfrees** [reply]
The binary is already uploading there since this morning :)
Since the speed is about three times what launchpad permits there's a chance everything may be published at last this afternoon

**Reply by Noordfrees** [reply]
Took only 5½ hours or so. I think we'll use GitHub only and no more launchpad for the final release :)

**Reply by TinoM** [reply]
I've uploaded the 32bit installer to launchpad and macos dmg files /64bit to github. I hope this is okay...
